lib/ansible/plugins/callback/grafana_annotations.py

```diff
@@ -249,15 +249,28 @@ def v2_runner_on_failed(self, result, **kwargs):
             'tags': ['ansible', 'ansible_event_failure', self.playbook]
         }
         self.errors += 1
+        self._send_annotations(data)
+
+    def _send_annotations(self, data):
+        uses_panels = False
         if self.dashboard_id:
             data["dashboardId"] = int(self.dashboard_id)
         if self.panel_id:
-            data["panelId"] = int(self.panel_id)
-        self._send_annotation(json.dumps(data))
+            uses_panels = True
```

Comment:
I think a function handling the `self.panel_id` would help make the code clearer.

The function could always return a list of integer or an empty list.
Then you can iterate on the list to send the annotation. When the list is empty nothing would be done and the check line 268 is no more required.

Reply:
Thank you for your suggestion. Its done. Do you know how to fix this deprecation warning?

> [DEPRECATION WARNING]: grafana_annotations callback, does not support setting 'options', it will work for now,  but this will be required in the future and should be updated,  see the 2.4 porting guide for details.. This feature will be removed in version 2.9. Deprecation warnings can be disabled by setting  deprecation_warnings=False in ansible.cfg.

Reply:
no, I forgot this was around ...
I think I'll ask for tips on IRC

Reply:
Got it: https://github.com/ansible/ansible/pull/57629
My best patch ever :)